lib/mp4/transmuxer.js

```diff
@@ -132,7 +132,7 @@ AudioSegmentStream = function(track, options) {
   };
 
   this.setEarliestDts = function(earliestDts) {
-    earliestAllowedDts = earliestDts - track.timelineStartInfo.baseMediaDecodeTime;
+    earliestAllowedDts = earliestDts - videoBaseMediaDecodeTime;
```

Comment:
To me this seemed like a better solution than using the `timelinestartinfo.baseMediaDecodeTime` because this is directly related to the current video `baseMediaDecodeTime` and not the initial `baseMediaDecodeTime`. 

Said another way it is used for trimming audio before the video start. We want to use an accurate `baseMediaDecodeTime` regardless of wether `keepOriginalTimestampOffsets` is `true` or `false` and the only way to do that is to use `videoBaseMediaDecodeTime`.

Reply:
Because this is now using the `baseMediaDecodeTime` for a given segment, wouldn't this mean that any audio preceding video in the segment would be trimmed? Meaning that if a previous segment had an audio end shorter than video end, and the current segment fills that audio gap by having an audio start before video start, this would trim those audio frames and leave a gap in the audio buffer.

Reply:
You could be right hmmm, maybe this fix is going to be a bit more involved.